test/integration/puppetlabs/services/jruby/service_macros_test.clj

```diff
@@ -4,21 +4,21 @@
             [puppetlabs.services.jruby.jruby-puppet-service :as jruby-puppet]
             [puppetlabs.services.puppet-profiler.puppet-profiler-service :as profiler]
             [puppetlabs.trapperkeeper.testutils.bootstrap :as tk-testutils]
-            [puppetlabs.services.jruby-pool-manager.jruby-pool-manager-service :as jruby-utils]
             [puppetlabs.services.jruby.jruby-puppet-testutils :as jruby-testutils]))
 
 (deftest simple-with-lock-test
   (testing "using with-lock doesn't throw any errors"
-    (tk-testutils/with-app-with-config
-     app
-     [profiler/puppet-profiler-service
-      jruby-puppet/jruby-puppet-pooled-service
-      jruby-utils/jruby-pool-manager-service]
-     (jruby-testutils/jruby-puppet-tk-config
-      (jruby-testutils/jruby-puppet-config))
-     (let [jruby-service (tk-app/get-service app :JRubyPuppetService)]
-       (jruby-puppet/with-lock
-        jruby-service :with-lock-test
-        ; Just make sure the macro doesn't fail
-        (is (true? true)))
-       (is (true? true))))))
+    (let [config (jruby-testutils/jruby-puppet-tk-config
+                  (jruby-testutils/jruby-puppet-config))]
+      (tk-testutils/with-app-with-config
+       app
+       [profiler/puppet-profiler-service
+        jruby-puppet/jruby-puppet-pooled-service
+        (jruby-testutils/mock-jruby-pool-manager-service config)]
```

Comment:
The locking controls whether borrows can be done against a `JRubyPool` object.  Even in the case of mocking, a real `JRubyPool` object is still created and exercised when the lock is performed.  The thing that's different when the mock is in place is what the objects in the pool are - mock `JRubyPuppet` objects.  If you're concerned enough about this, I could convert it back to using the real pool manager.  I don't think it's necessary, though, to exercise the code that we're trying to test here.